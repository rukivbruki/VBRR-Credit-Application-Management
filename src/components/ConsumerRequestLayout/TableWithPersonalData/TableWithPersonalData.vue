<template>
  <el-skeleton :loading="getIsLoadingPersonalData" animated>
    <template #template>
      <el-table :data="[{}]" border style="width: 100%">
        <el-table-column
          v-for="{ label } of headerList"
          :key="label"
          :label="label"
        >
          <template #default>
            <el-skeleton-item variant="text" />
          </template>
        </el-table-column>
      </el-table>
    </template>
    <template #default>
      <el-table
        ref="singleTable"
        :data="croppedPersonList"
        highlight-current-row
        @current-change="handleCurrentChange"
        style="width: 100%; margin-top: 80px"
        stripe
        border
        empty-text="Данные отсутствуют, измените параметра поиска"
        @row-dblclick="dblClickOnTableRow"
      >
        <el-table-column
          v-for="{ label, prop } of headerList"
          :key="label"
          :property="prop"
          :label="label"
        >
          <template #header v-if="label === headerList[0].label">
            <section class="search">
              <span>{{ headerList[0].label }}</span>
              <el-input
                class="search__input"
                v-model="search"
                size="large"
                placeholder="Поиск по имени"
              />
            </section>
          </template>
        </el-table-column>
      </el-table>
    </template>
  </el-skeleton>
  <Pagination :numberOfPages="this.getPersonListLength"></Pagination>
</template>

<script>
import Pagination from '../Pagination/Pagination';
import { mapGetters } from 'vuex';

export default {
  name: 'TableWithPersonalData',
  components: { Pagination },
  props: ['isSelectBtnClicked'],
  data() {
    return {
      currentRow: null,
      headerList: [
        {
          label: 'ФИО',
          prop: 'personInfo.fullName',
        },
        {
          label: 'Дата рождения',
          prop: 'personInfo.birthDay',
        },
        {
          label: 'Тип документа',
          prop: 'personInfo.paperInfo.paperTypeString',
        },
        {
          label: 'Серия документа',
          prop: 'personInfo.paperInfo.paperSeries',
        },
        {
          label: 'Номер документа',
          prop: 'personInfo.paperInfo.paperNumber',
        },
        {
          label: 'Адрес прописки',
          prop: 'personInfo.addressInfo.fullAddress',
        },
      ],
      search: '',
    };
  },
  computed: {
    ...mapGetters([
      'getCroppedPersonList',
      'getPersonListLength',
      'getIsLoadingPersonalData',
    ]),
    croppedPersonList() {
      return this.getCroppedPersonList.filter(
        (data) =>
          !this.search ||
          data.personInfo.fullName
            .toLowerCase()
            .includes(this.search.toLowerCase()),
      );
    },
  },
  methods: {
    handleCurrentChange(row) {
      this.$emit('select-person-id', row.personId);
    },
    dblClickOnTableRow(row) {
      this.$router.push({
        path: `/consumer-request/${row.personId}/info`,
      });
    },
  },
};
</script>

<style>
.search {
  display: flex;
  align-items: center;
}
.search__input {
  padding-left: 10px;
}
</style>
