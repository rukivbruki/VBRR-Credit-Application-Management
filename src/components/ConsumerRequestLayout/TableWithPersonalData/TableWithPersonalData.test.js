import { mount } from '@vue/test-utils';
import TableWithPersonalData from './TableWithPersonalData';

describe('TableWithPersonalData.vue render', () => {
  test('renders correctly', () => {
    expect(mount(TableWithPersonalData).element).toMatchSnapshot();
  });
});
