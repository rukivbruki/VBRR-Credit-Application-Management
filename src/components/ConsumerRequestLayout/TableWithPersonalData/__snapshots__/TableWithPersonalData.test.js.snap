// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableWithPersonalData.vue render renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <el-table
    data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    highlight-current-row=""
    style="width: 100%;"
  >
    <el-table-column
      type="index"
      width="50"
    />
    <el-table-column
      label="ФИО"
      property="name"
    />
    <el-table-column
      label="Дата рождения"
      property="date"
    />
    <el-table-column
      label="Тип документа"
      property="typeOfDocument"
    />
    <el-table-column
      label="Серия документа"
      property="documentSeries"
    />
    <el-table-column
      label="Номер документа"
      property="documentNumber"
    />
    <el-table-column
      label="Адресс прописки"
      property="address"
    />
  </el-table>
  <div
    class="block"
  >
    <el-pagination
      background=""
      layout="prev, pager, next"
      total="20"
    />
  </div>
  
</div>
`;
